### PR TITLE
Fix: Resolve issue where mixed-case stream properties would result in missing data

### PR DESCRIPTION
**Pull request by aaronsteers**
Resolves: #108

The behavior implemented is as follows:

- Prefer lower case column names in tables, as before.
- Expect mixed-case property names in the incoming data from the source. 

On its own this creates a small problem, which was that accessing the results of Source.get_records() would require different casing than was required to read from data synced to the cache with Source.read().

The way I was able to resolve this difference, is to introduce a case-insensitive dictionary implementation - which as of now is only implemented in the output of Source.get_records().

This mirrors the database behavior of (1) preferring lower-casing for column names, while (2) still allowing case-insensitive matching by default.

### CODE REVIEW

File: airbyte/sources/base.py [hunk 1]
```diff
@@ -329,29 +329,21 @@ def get_records(self, stream: str) -> LazyDataset:
...
 
         def _with_logging(records: Iterable[dict[str, Any]]) -> Iterator[dict[str, Any]]:
             self._log_sync_start(cache=None)
             yield from records
             self._log_sync_success(cache=None)
 
-        def _with_missing_columns(records: Iterable[dict[str, Any]]) -> Iterator[dict[str, Any]]:
-            """Add missing columns to the record with null values."""
-            for record in records:
-                existing_properties_lower = lower_case_set(record.keys())
```
Comment (aaronsteers) [inline airbyte/sources/base.py:342]: This "missing columns" treatment is now included in the new "normalize_records" function.

File: tests/integration_tests/fixtures/source-test/source_test/run.py [hunk 2]
```diff
@@ -17,8 +17,8 @@
                     "$schema": "http://json-schema.org/draft-07/schema#",
                     "type": "object",
                     "properties": {
-                        "column1": {"type": "string"},
-                        "column2": {"type": "number"},
+                        "Column1": {"type": "string"},
+                        "Column2": {"type": "number"},
```
Comment (aaronsteers) [inline tests/integration_tests/fixtures/source-test/source_test/run.py:21]: Modified the test data to ensure coverage of mixed-case top level properties.

File: airbyte/_processors/sql/duckdb.py [hunk 3]
```diff
@@ -93,9 +94,9 @@ def _write_files_to_new_table(
         columns_type_map = indent(
             "\n, ".join(
                 [
-                    f"{self._quote_identifier(c)}: "
-                    f"{self._get_sql_column_definitions(stream_name)[c]!s}"
-                    for c in columns_list
+                    f"{self._quote_identifier(prop_name)}: "
+                    f"{self._get_sql_column_definitions(stream_name)[prop_name.lower()]!s}"
```
Comment (bindipankhudi) [inline airbyte/_processors/sql/duckdb.py]: is there a reason we are not using the LowercaseNormalizer class here and doing .lower()?

**Review by bindipankhudi — APPROVED** [review]
Looks great!

File: airbyte/_processors/sql/duckdb.py (hunk 3 above)
Comment (aaronsteers) [reply airbyte/_processors/sql/duckdb.py]: Good catch! No reason. There are probably a few more cases like this actually, where we can standardize the logic in one place.

An advantage of doing so will be that later, more edge cases can be all handled in one place - for instance, handling of punctuation and special characters.

Comment (aaronsteers) [reply airbyte/_processors/sql/duckdb.py]: Fixed now. ✅

**Pull request merged**